core/src/main/resources/error/error-classes.json

```diff
@@ -1539,6 +1539,13 @@
       "Parse Mode: <failFastMode>. To process malformed records as null result, try setting the option 'mode' as 'PERMISSIVE'."
     ]
   },
+  "MERGE_CARDINALITY_VIOLATION" : {
+    "message" : [
+      "The ON search condition of the MERGE statement matched a single row from the target table with multiple rows of the source table.",
+      "This could result in the target row being operated on more than once with an update or delete operation and is not allowed."
+    ],
+    "sqlState" : "23K01"
```

Comment:
Thank you for updates. We can update `README.txt` later. (https://github.com/apache/spark/pull/41448#discussion_r1222454470)

Reply:
Done.